x/concentrated-liquidity/incentives.go

```diff
@@ -970,8 +975,16 @@ func (k Keeper) claimAllIncentivesForPosition(ctx sdk.Context, positionId uint64
 
 				var forfeitedIncentivesPerShare sdk.DecCoins
 				for _, coin := range collectedIncentivesForUptime {
-					// updated forfeitedIncentivesPerShare to add back = collectedIncentivesPerShare / totalSharesAccum
-					forfeitedIncentivesPerShare = append(forfeitedIncentivesPerShare, sdk.NewDecCoinFromDec(coin.Denom, coin.Amount.ToDec().Add(dust.AmountOf(coin.Denom)).Quo(totalSharesAccum)))
+					// updated forfeitedIncentivesPerShare to add back = collectedIncentivesPerShare / (totalSharesAccum - position.NumShares)
+					denominator := totalSharesAccum.Sub(pos.NumShares)
```

Comment:
Can you clarify why this is the correct value here? Intuitively, it seems like the calculation should be:

1. `forfeitedIncentivesPerShare * pos.NumShares`, which gives us the total forfeited incentives in DecCoins
2. Result from above divided by the shares on the current tick, since those are the only shares that will qualify for these rewards anyway

In other words, this bug seems to be an extension of the earlier incentives distribution bug, so we should expect the solution to look somewhat similar as well.

With this setup, I could create a 1-tick wide position with $1M at `maxTick` and ensure that majority of forfeited incentives are never claimable (by crowding out the denominator in your calculation)

Reply:
To clarify, this should likely include qualifying balancer shares as well, similar to the logic here: https://github.com/osmosis-labs/osmosis/blob/2fd041531dcb995f7263101d2d3b0901f1c6445a/x/concentrated-liquidity/incentives.go#L404

It might be worth abstracting this logic into a helper to ensure it is consistent (`GetQualifyingShares`)